src/search/sql_transformer.h

```diff
@@ -57,7 +58,42 @@ StatusOr<std::unique_ptr<parse_tree::node>> ParseToTree(Input&& in) {
 struct Transformer : ir::TreeTransformer {
   explicit Transformer(const ParamMap& param_map) : TreeTransformer(param_map) {}
 
+  StatusOr<std::unique_ptr<VectorLiteral>> Transform2Vector(const TreeNode& node) {
+    std::string vector_str;
+    if (Is<StringL>(node)) {
+      vector_str = GET_OR_RET(UnescapeString(node->string_view()));
+    } else if (Is<Param>(node)) {
+      vector_str = GET_OR_RET(GetParam(node));
+    } else {
+      return {Status::NotOK, "encountered invalid vector input"};
+    }
+
+    if (vector_str.size() < 2 || vector_str[0] != '[' || vector_str.back() != ']') {
+      return {Status::NotOK, "encountered invalid vector input without brackets"};
+    }
+
+    std::vector<double> values = GET_OR_RET(ParseFloatArray(vector_str.substr(1, vector_str.size() - 2)));
```

Comment:
I believe we should standardize the parameter format here. 

Regardless of the frontend language used, values of vector parameters should be represented in binary format.

This will make it easier to understand for users.